### ISSUE TITLE
Kael'thas Sunstrider, Magisters' Terrace

### ISSUE THREAD

**Issue by Tafuna** [scriptdev2/scriptdev2-tbc#20]
When he pushs u away into the air u don't stay up in the air and u can't fly.
u just stay on the ground

Im sure thats has worked some month ago but now is broken BUT i tested the spell only and the spell when i cast self on me works

**Comment by vovk** [comment]
Something wrong with wright or read caster guid... Try to cast Gravity Lapse on some one... you will cast it on self. Same thing with Kaeltath, he cast it on self, so gravity lapse fly not casts at all, because he cast it on self & he isn't player.
The same shit with many other spells for now... hunter traps(Immolation Trap become infinity), electric storm(Zul'Aman Akil'Zon) & so on... And i missed that commit, whitch broke this :(

**Comment by Tafuna** [comment]
yeah i saw a lot of spells a broken what worked befor fine, i check few core commits too but not found the commit what will break the spells

**Comment by evil-at-wow** [comment]
@Tafuna @vovk 
If you happen to have some time to test, could you do me a favor with a fairly quick test? You should be able to go back to revision [s2269] - cmangos/mangos-tbc@476af655f323debf87df3c17dbbba135cac3496b without having to change your scriptdev2 revision and without requiring any database changes. Can you do a git checkout of that core revision, then rebuild the core and test if the spells you are talking about are still bugged, please?

**Comment by vovk** [comment]
@evil-at-wow  sorry i wrong read issue :(
i compile with 2277 rev. earlier bacause sd2 compiles with s2277 not with s2269 as there written. And compile last rev. Test him twice. He cast both spells on target. They just didn't stuck.44226 & 44227. They never stuck before as i remember. In core should be added exception.

About that spells that i wrote. Maybe i need to start new issue? They have all same thing - they all triggered.
1. Zul'Aman Akil'Zon casts Electrical Storm 43648 on target. Aplly aura SPELL_AURA_PERIODIC_TRIGGER_SPELL. This aura trigger 43658 spell. It should be casted by target on self. But for some reason <non-existing source> cast 43658 spell on Akil'Zon.
2. Sunwell Kalecgos cast Spectral Blast 44869 on one target. This spell trigger 44866, 46648 & 44811. If i had only two players one tank(main agro) & some one else who hit him, kelecgos after 30 seconds should place second player into spectral realm, but it not happes because he can't find target. If on Kalecgos 10+ people all works seems to be ok, because TargetMap big, not just one player & this bug appear infrequently.
3. Hunter Immolation Trap. On trigger should dissapear. For now it trigger on target & stay, it target gets on this trap again, it triggers second time & so on.
That's what I thought at once. Later i could remember more.

**Comment by xfurry** [comment]
Just to be clear from my side: On all these issues that are opened on sd2-tbc, are we really talking about the TBC client & TBC DB, or is it just about the instance and the issue is confirmed in 335a as well?

**Comment by Tafuna** [comment]
i self test on tbc but most of this is on wotlk issues too
